### PR TITLE
Unifiy the creation of photo links

### PR DESCRIPTION
**Pull request by annando**
We have got two different formats for picture links with previews. They are now created in central functions.

### CODE REVIEW

File: src/Util/Images.php [hunk 1]
```diff
@@ -316,4 +316,32 @@ public static function getScalingDimensions(int $width, int $height, int $max):
...
+
+	/**
+	 * Get a link to a picture with a preview
+	 *
+	 * @param string $photo
+	 * @param string $preview
+	 * @param string $description
+	 * @return string
+	 */
+	public static function getPictureUrl(string $photo, string $preview, string $description): string
```
Comment (MrPetovan) [inline src/Util/Images.php]:
```suggestion
	 * Get a BBCode tag for an local photo page URL with a preview thumbnail and an image description
	 *
	 * @param string $resource_id
	 * @param string $nickname The local user owner of the resource
	 * @param int    $preview Preview image size identifier, either 0, 1 or 2 in decreasing order of size
	 * @param string $ext Image file extension
	 * @param string $description
	 * @return string
	 */
	public static function getBBCodeByResource(string $resource_id, string $nickname, int $preview, string $ext, string $description): string
	{
		return $this->getBBCodeByUrl(
			DI::baseUrl() . '/photos/' . $nickname . '/image/' . $resource_id,
			DI::baseUrl() . '/photo/' . $resource_id . '-' . $preview. '.' . $ext,
			$description
		);
	}

	/**
	 * Get a BBCode tag for an image URL with a preview thumbnail and an image description
	 *
	 * @param string $photo Full image URL
	 * @param string $preview Preview image URL
	 * @param string $description
	 * @return string
	 */
	public static function getBBCodeByUrl(string $photo, string $preview, string $description): string
```

**Review by MrPetovan — CHANGES_REQUESTED** [review]

File: src/Util/Images.php [hunk 2]
```diff
@@ -316,4 +316,38 @@ public static function getScalingDimensions(int $width, int $height, int $max):
...
+
+	/**
+	 * Get a BBCode tag for an image URL with a preview thumbnail and an image description
+	 *
+	 * @param string $photo Full image URL
+	 * @param string $preview Preview image URL
+	 * @param string $description
+	 * @return string
+	 */	public static function getBBCodeByUrl(string $photo, string $preview = null, string $description = null): string
+	{
```
Comment (MrPetovan) [inline src/Util/Images.php]: The description is always used in string concatenation.
```suggestion
	 */
	public static function getBBCodeByUrl(string $photo, string $preview = null, string $description = ''): string
	{
```

**Review by MrPetovan — CHANGES_REQUESTED** [review]

**Review by MrPetovan — APPROVED** [review]

**Comment by MrPetovan** [comment]
From the PHPUnit log:
```
1) Friendica\Test\src\Module\Api\Twitter\Statuses\UpdateTest::testApiStatusesUpdate
TypeError: Argument 3 passed to Friendica\Util\Images::getBBCodeByUrl() must be of the type string, null given, called in /woodpecker/src/github.com/friendica/friendica/src/Model/Post/Media.php on line 1007

/woodpecker/src/github.com/friendica/friendica/src/Util/Images.php:347
/woodpecker/src/github.com/friendica/friendica/src/Model/Post/Media.php:1007
/woodpecker/src/github.com/friendica/friendica/src/Factory/Api/Twitter/Status.php:150
/woodpecker/src/github.com/friendica/friendica/src/Factory/Api/Twitter/Status.php:113
/woodpecker/src/github.com/friendica/friendica/src/Module/Api/Twitter/Statuses/Update.php:186
/woodpecker/src/github.com/friendica/friendica/src/BaseModule.php:228
/woodpecker/src/github.com/friendica/friendica/src/Module/BaseApi.php:105
/woodpecker/src/github.com/friendica/friendica/tests/src/Module/Api/Twitter/Statuses/UpdateTest.php:59
```

**Comment by annando** [comment]
@MrPetovan yeah, your suggestion reverted some change that I made to make the tests happy :-)

File: src/Model/Post/Media.php [hunk 3]
```diff
@@ -1004,19 +1004,7 @@ public static function addAttachmentsToBody(int $uriid, string $body = '', array
...
-						$body .= "\n[url=" . $media['url'] . "][img]" . $media['preview'] . '[/img][/url]';
-					}
-				} else {
-					if (!empty($media['description'])) {
-						$body .= "\n[img=" . $media['url'] . ']' . $media['description'] . '[/img]';
-					} else {
-						$body .= "\n[img]" . $media['url'] . '[/img]';
-					}
-				}
+				$body .= "\n" . Images::getBBCodeByUrl($media['url'], $media['preview'], $media['description']);
```
Comment (MrPetovan) [inline src/Model/Post/Media.php]:
```suggestion
				$body .= "\n" . Images::getBBCodeByUrl($media['url'], $media['preview'], $media['description'] ?? '');
```

File: src/Util/Images.php [hunk 4]
```diff
@@ -316,4 +316,39 @@ public static function getScalingDimensions(int $width, int $height, int $max):
...
+
+	/**
+	 * Get a BBCode tag for an image URL with a preview thumbnail and an image description
+	 *
+	 * @param string $photo Full image URL
+	 * @param string $preview Preview image URL
+	 * @param string $description
+	 * @return string
+	 */
+	public static function getBBCodeByUrl(string $photo, string $preview = null, string $description = null): string
```
Comment (MrPetovan) [inline src/Util/Images.php]: While it's a quick fix I don't think we should have the description nullable, and instead fix the offending call.
```suggestion
	public static function getBBCodeByUrl(string $photo, string $preview = null, string $description = ''): string
```

**Review by MrPetovan — CHANGES_REQUESTED** [review]

File: src/Util/Images.php [hunk 5]
```diff
@@ -347,8 +347,17 @@ public static function getBBCodeByResource(string $resource_id, string $nickname
 	public static function getBBCodeByUrl(string $photo, string $preview = null, string $description = null): string
 	{
 		if (!empty($preview)) {
-			return '[url=' . $photo . '][img=' . $preview . ']' . $description . '[/img][/url]';
+			if (!is_null($description)) {
+				return '[url=' . $photo . '][img=' . $preview . ']' . $description . '[/img][/url]';
+			} else {
+				return '[url=' . $photo . '][img]' . $preview . '[/img][/url]';
```
Comment (MrPetovan) [inline src/Util/Images.php]: We've had a discussion about how the default generated BBCode for inserted images was cumbersome to edit to allow for image descriptions, please revert to the previous version where `$description` is never expected to be null and we don't use the `[img]Image URL[/img]` form anymore, only the `[img=Image URL][/img]` form.

**Review by MrPetovan — CHANGES_REQUESTED** [review]

File: src/Util/Images.php (hunk 5 above)
Comment (annando) [reply src/Util/Images.php]: See https://github.com/friendica/friendica/pull/13107/files#diff-d87c4b23232100f153fcbb4d5f327ba0882b94ca767226103ca2fb44c5c37385R182 - at upload (where it matters), the format with the easy way to enter a description will be chosen.

Comment (MrPetovan) [reply src/Util/Images.php]: Yes, but we never need the shorter format, which would simplify the method you created. We never need an actual distinction between not having a description and having an empty description.

**Comment by MrPetovan** [comment]
I've seen you updated the branch to the latest develop, please simplify the method by removing the case without image description.

**Comment by annando** [comment]
May I drive your attention to the second commit here?
image

**Review by MrPetovan — APPROVED** [review]

**Pull request merged**